x/ecocredit/migrations/v3/core.go

```diff
@@ -0,0 +1,404 @@
+package v3
+
+import (
+	"context"
+	"sort"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	ormerrors "github.com/cosmos/cosmos-sdk/orm/types/ormerrors"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
+	orm "github.com/regen-network/regen-ledger/orm"
+)
+
+type batchMapT struct {
+	Id              uint64
+	AmountCancelled string
+}
+
+// MigrateState performs in-place store migrations from v3.0 to v4.0.
+func MigrateState(sdkCtx sdk.Context, storeKey storetypes.StoreKey,
```

Comment:
should we also include the params migration here? that would close #965. It should just take the existing params, convert them to v1 proto, and then set the default AskDenom param (https://github.com/regen-network/regen-ledger/pull/969).

Reply:
> should we also include the params migration here? that would close #965. It should just take the existing params, convert them to v1 proto, and then set the default AskDenom param (#969).

We don't have any breaking changes in params (It is just a version bump). For `allowed_ask_denoms` we can set them in the upgrade handler.